src/models/emotion_detection/samo_bert_emotion_classifier.py

```
@@ -0,0 +1,520 @@
+        # Pass through classification head
+        logits = self.classifier(pooled_output)
+
+        # Apply temperature scaling
+        logits = logits / self.temperature
+
+        return logits
+
+    def predict_emotions(
+        self,
+        texts: Union[str, List[str]],
+        threshold: float = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)